lucid/misc/io/loading.py

```diff
@@ -204,20 +249,27 @@ def load_using_loader(url_or_handle, loader, cache, **kwargs):
             # since this may have been cached, it's our responsibility to try again once
             # since we use a handle here, the next DecodeError should propagate upwards
             with read_handle(url, cache="purge") as handle:
-                result = load_using_loader(handle, loader, cache, **kwargs)
+                result = load_using_loader(handle, decompressor, loader, cache, **kwargs)
     return result
 
 
 def is_handle(url_or_handle):
     return hasattr(url_or_handle, "read") and hasattr(url_or_handle, "name")
 
 
-def get_extension(url_or_handle):
+def _get_extension(url_or_handle):
+    compression_ext = None
     if is_handle(url_or_handle):
-        _, ext = os.path.splitext(url_or_handle.name)
+        path_without_ext, ext = os.path.splitext(url_or_handle.name)
+    else:
+        path_without_ext, ext = os.path.splitext(url_or_handle)
+
+    if ext in decompressors:
+        decompressor_ext = ext
+        _, ext = os.path.splitext(path_without_ext)
     else:
-        _, ext = os.path.splitext(url_or_handle)
+        decompressor_ext = None
```

Comment:
(no action needed) It seems like unsupported decompressors (eg. ".gz") will fall through as normal extensions. I don't see any great way to avoid this, but it may lead to unintuitive error messages.

Reply:
I agree, I wasn't sure if I should try to detect the "two or more dots" in filename case and treat it in a special way. Maybe I can add something about compressors to the error message.